### PR TITLE
Add eslint on the default BO theme

### LINKED ISSUE #22998: Default theme is the only folder not having ESLint installed and configured
#### Is your feature request related to a problem?

As we wants to increase the quality of the codebase of our software, default theme is missing ESLint for code quality/style, we should add it

#### Describe the solution you'd like

Adding ESLint on default theme and fixing as much errors as we can

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We wants to install and fix all eslint errors on default theme
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22998.
| How to test?      | Introspect every legacy pages, that touch JS, so it's all about behaviors of JavaScript side
| Possible impacts? | Every legacy page, including dashboard

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22999)

### CODE REVIEW

File: admin-dev/themes/default/js/bundle/admin_parameters/performancePage.js [hunk 1]
```diff
@@ -22,98 +22,100 @@
...
 
-        return {
-            'server_ip': serverIpInput.value,
-            'server_port': serverPortInput.value,
-            'server_weight': serverWeightInput.value,
-        };
+    return {
+      server_ip: serverIpInput.value,
+      server_port: serverPortInput.value,
+      server_weight: serverWeightInput.value,
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/admin_parameters/performancePage.js]:
```suggestion
      server_ip: document.getElementById('memcache_ip').value,
      server_port: document.getElementById('memcache_port').value,
      server_weight: document.getElementById('memcache_weight').value,
```

File: admin-dev/themes/default/js/bundle/admin_parameters/performancePage.js [hunk 2]
```diff
@@ -22,98 +22,100 @@
...
+  };
 
-    this.getFormValues = function() {
-        var serverIpInput = document.getElementById('memcache_ip');
-        var serverPortInput = document.getElementById('memcache_port');
-        var serverWeightInput = document.getElementById('memcache_weight');
+  this.getFormValues = function () {
+    const serverIpInput = document.getElementById('memcache_ip');
+    const serverPortInput = document.getElementById('memcache_port');
+    const serverWeightInput = document.getElementById('memcache_weight');
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/admin_parameters/performancePage.js]:
```suggestion
```

File: admin-dev/themes/default/js/bundle/module/module.js [hunk 3]
```diff
@@ -263,54 +260,59 @@ var AdminModuleController = function() {
...
       if (key.split('-').length > 1) {
         key = key.split('-')[0];
       }
-      if (self.currentSorting.indexOf('-desc') != -1) {
+      if (self.currentSorting.indexOf('-desc') !== -1) {
         order = 'desc';
       }
 
+      // eslint-disable-next-line
       function currentCompare(a, b) {
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/module/module.js]: Like it's a private method, you can adapt it for fixing eslint

File: admin-dev/themes/default/js/bundle/pagination.js [hunk 4]
```diff
@@ -23,56 +23,65 @@
...
+        $(this).val(max);
+        return false;
+      }
+    });
+
+    // eslint-disable-next-line
+    $(this).on('blur', function (e) {
+      const val = parseInt($(e.target).val(), 10);
+
+      if (parseInt(val, 10) > 0) {
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/pagination.js:79]:
```suggestion
      if (val > 0) {
```

File: admin-dev/themes/default/js/bundle/product/default-category.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
-  var defaultCategoryForm = $('#form_step1_id_category_default');
+const defaultCategory = (function () {
+  const defaultCategoryForm = $('#form_step1_id_category_default');
+
   return {
-    'init': function () {
-      /** Populate category tree with the default category **/
-      var defaultCategoryId = defaultCategoryForm.find('input:checked').val();
+    init() {
+      /** Populate category tree with the default category * */
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/default-category.js]:
```suggestion
      /** Populate category tree with the default category */
```

File: admin-dev/themes/default/js/bundle/right-sidebar.js [hunk 6]
```diff
@@ -6,143 +6,147 @@
...
+  $.fn.sidebar.Constructor = Sidebar;
+
+  $.fn.sidebar.noConflict = function () {
+    $.fn.sidebar = old;
+    return this;
+  };
+
+  $(document).on('click.bs.sidebar.data-api', '[data-toggle="sidebar"]', function (e) {
+    const $this = $(this); let
+      href;
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/right-sidebar.js]:
```suggestion
    const $this = $(this);
    let href;
```

File: admin-dev/themes/default/webpack.config.js [hunk 7]
```diff
@@ -23,67 +23,67 @@
...
-      './js/theme.js'
+      './js/theme.js',
     ],
     output: {
       path: path.resolve(__dirname, 'public'),
-      filename: 'bundle.js'
+      filename: 'bundle.js',
     },
-    //devtool: 'source-map', // uncomment me to build source maps (really slow)
+    // devtool: 'source-map', // uncomment me to build source maps (really slow)
```
Comment (Progi1984) [inline admin-dev/themes/default/webpack.config.js]:
```suggestion
```

File: admin-dev/themes/default/webpack.config.js [hunk 8]
```diff
@@ -23,67 +23,67 @@
...
       }, {
         test: /\.(scss|sass|css)$/,
         use: [
           MiniCssExtractPlugin.loader,
           {
             loader: 'css-loader',
             options: {
-              //sourceMap: true, // uncomment me to generate source maps
-            }
+              // sourceMap: true, // uncomment me to generate source maps
```
Comment (Progi1984) [inline admin-dev/themes/default/webpack.config.js]:
```suggestion
```

File: admin-dev/themes/default/webpack.config.js [hunk 9]
```diff
@@ -23,67 +23,67 @@
...
+            },
           },
           {
             loader: 'postcss-loader',
             options: {
-              //sourceMap: true, // uncomment me to generate source maps
-            }
+              // sourceMap: true, // uncomment me to generate source maps
```
Comment (Progi1984) [inline admin-dev/themes/default/webpack.config.js]:
```suggestion
```

File: admin-dev/themes/default/webpack.config.js [hunk 10]
```diff
@@ -23,67 +23,67 @@
...
+            },
           },
           {
             loader: 'sass-loader',
             options: {
-              //sourceMap: true, // uncomment me to generate source maps
-            }
-          }
-        ]
+              // sourceMap: true, // uncomment me to generate source maps
```
Comment (Progi1984) [inline admin-dev/themes/default/webpack.config.js]:
```suggestion
```

File: admin-dev/themes/default/js/tree.js [hunk 11]
```diff
@@ -22,253 +22,242 @@
...
+            }
 
-		return $(this);
-	},
+            const thatOne = $(this);
+            $.get(
+              'ajax-tab.php',
+              {
+                // eslint-disable-next-line
+                controller: 'AdminProducts', token: currentToken, action: 'getCategoryTree', type: idTree, category, inputName: name, useCheckBox,
```
Comment (Progi1984) [inline admin-dev/themes/default/js/tree.js]:
```suggestion
                controller: 'AdminProducts',
                token: currentToken,
                action: 'getCategoryTree',
                type: idTree,
                category,
                inputName: name,
                useCheckBox,
```

File: admin-dev/themes/default/js/tree.js [hunk 12]
```diff
@@ -22,253 +22,242 @@
...
+
+      if (inputType === 'checkbox') {
+        useCheckBox = 1;
+      }
+
+      $.get(
+        'ajax-tab.php',
+        {
+          // eslint-disable-next-line
+          controller: 'AdminProducts', token: currentToken, action: 'getCategoryTree', type: idTree, fullTree: 1, selected, inputName: name, useCheckBox,
```
Comment (Progi1984) [inline admin-dev/themes/default/js/tree.js]:
```suggestion
          controller: 'AdminProducts',
          token: currentToken,
          action: 'getCategoryTree',
          type: idTree,
          fullTree: 1,
          selected,
          inputName: name,
          useCheckBox,
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: admin-dev/themes/default/js/jquery.fileupload-validate.js [hunk 13]
```diff
@@ -9,48 +9,45 @@
  * http://www.opensource.org/licenses/MIT
  */
 
-/*jslint nomen: true, unparam: true, regexp: true */
-/*global define, window */
+/* jslint nomen: true, unparam: true, regexp: true */
```
Comment (PierreRambaud) [inline admin-dev/themes/default/js/jquery.fileupload-validate.js:12]: Woot, just notice about jslint :scream: 
I think you can exclude these dependencies, no?

Comment (Progi1984) [reply admin-dev/themes/default/js/jquery.fileupload-validate.js:12]: Or remove dependencies and add it in package.json ?

Comment (NeOMakinG) [reply admin-dev/themes/default/js/jquery.fileupload-validate.js:12]: I'll remove it

Comment (NeOMakinG) [reply admin-dev/themes/default/js/jquery.fileupload-validate.js:12]: Oh, I just noticed it's a lib, we shouldn't modify anything to this file!

**Comment by PierreRambaud** [comment]
You remove the WIP label, but a wild conflict appears :scream:

**Comment by Progi1984** [comment]
@NeOMakinG Need rebase

**Review by PierreRambaud — DISMISSED** [review]

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 14]
```diff
@@ -289,176 +305,178 @@ var formCategory = (function() {
...
-            '</label>' +
-          '</div>' +
-          '</li>';
-
-        var parentElement = $('#form_step1_categories input[value='+response.category.id_parent+']').parent().parent();
-        if(parentElement.next('ul').length === 0){
-          html = '<ul>' + html + '</ul>';
+      success(response) {
+        // inject new category into category tree
+        let html = `${'<li>'
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js]:
```suggestion
        let html = `${'<li>'
```

Where is the end of `${` ?

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 15]
```diff
@@ -536,135 +555,146 @@ var stock = (function() {
...
 
-    if (target == '_blank' && redirect) {
-      var openBlank = window.open('about:blank', target, '');
+    if (target === '_blank' && redirect) {
+      openBlank = window.open('about:blank', target, '');
       openBlank.document.write(
-        '<p style="text-align: center;">' +
-        '<img src="' + document.location.origin + baseAdminDir + '/themes/default/img/spinner.gif">' +
-        '</p>'
+        `${'<p style="text-align: center;">'
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js]:
```suggestion
        `${'<p style="text-align: center;">'
```

Where is the end of `${` ?

File: admin-dev/themes/default/js/tree.js [hunk 16]
```diff
@@ -22,253 +22,253 @@
...
+          selected.push($(this).val());
+        },
+      );
+      const name = $(`#${idTree}`).find('ul.tree input').first().attr('name');
+      const inputType = $(`#${idTree}`).find('ul.tree input').first().attr('type');
+      let useCheckBox = 0;
+
+      if (inputType === 'checkbox') {
+        useCheckBox = 1;
+      }
```
Comment (Progi1984) [inline admin-dev/themes/default/js/tree.js]:
```suggestion
      const useCheckBox = inputType === 'checkbox' ? 1 :  0;
```

File: admin-dev/themes/default/js/tree.js [hunk 17]
```diff
@@ -22,253 +22,253 @@
...
-				$('.tree-selected').removeClass("tree-selected");
-				$('li input:checked').parent().addClass("tree-selected");
-			}
-		);
+          if (loadTree) {
+            const category = $(this).parent().children('ul.tree input').first()
+              .val();
+            const inputType = $(this).parent().children('ul.tree input').first()
+              .attr('type');
+            let useCheckBox = 0;
```
Comment (Progi1984) [inline admin-dev/themes/default/js/tree.js]:
```suggestion
      const useCheckBox = inputType === 'checkbox' ? 1 :  0;
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by aDamelincourt** [comment]
@NeOMakinG 
On dashboard, I've got these errors : 

image

image

Is it related to this PR ?

**Comment by NeOMakinG** [comment]
@aDamelincourt not related, related to a module or a browser extension

**Comment by aDamelincourt** [comment]
Thanks @NeOMakinG 
✔️ , check by QA, I can't find any error on legacies pages

**Comment by Progi1984** [comment]
@aDamelincourt I suppose the issue closing is an error.

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @aDamelincourt

**Pull request merged**